azure/functions/_authenticationevents.py

```diff
@@ -0,0 +1,214 @@
+                mail:str,
+                onPremisesSamAccountName:str,
+                onPremisesSecurityIdentifier:str,
+                onPremiseUserPrincipalName:str,
+                preferredDataLocation:str,
+                preferredLanguage:str,
+                surname:str,
+                userPrincipalName:str,
+                userType:str,
+                id:str):
+                self.id=id
+                self.userType=userType
+                self.userPrincipalName=userPrincipalName
+                self.surname=surname
+                self.preferredLanguage=preferredLanguage
+                self.ageGroup=ageGroup
+                self.companyName=companyName
+                self.country=country
+                self.createdDateTime=createdDateTime
+                self.creationType=creationType
+                self.department=department
+                self.displayName=displayName
+                self.givenName=givenName
+                self.lastPasswordChangeDateTime=lastPasswordChangeDateTime
+                self.mail=mail
+                self.onPremisesSamAccountName=onPremisesSamAccountName
+                self.onPremisesSecurityIdentifier=onPremisesSecurityIdentifier
+                self.onPremiseUserPrincipalName=onPremiseUserPrincipalName
+                self.preferredDataLocation=preferredDataLocation
+
+    def populate(user: dict):
+        return User(**user)
+
+                
+
+Roles=list[Role]
+
+
+class Context():
+    def __init__(self,
+                correlationId:str,
+                client:Client,
+                authProtocol:AuthProtocol,
+                clientServicePrincipal: ServicePrincipal,
+                resourceServicePrincipal: ServicePrincipal,
+                roles: Roles,
+                user: User):
+                self.user=user
+                self.roles=roles
+                self.resourceServicePrincipal=resourceServicePrincipal
+                self.clientServicePrincipal=clientServicePrincipal
+                self.authProtocol=authProtocol
+                self.client=client
+                self.correlationId=correlationId
+    
+    def populate(context: dict):
+        return Context(correlationId=context.get('correlationId'),
+        user=User.populate(context.get('user')),
+        client=Client.populate(context.get('client')),
+        clientServicePrincipal=ServicePrincipal.populate(context.get('clientServicePrincipal')),
+        resourceServicePrincipal=ServicePrincipal.populate(context.get('resourceServicePrincipal')),
+        roles=context.get('roles'),
+        authProtocol=AuthProtocol.populate(context.get('authProtocol')))
+
+class preview_10_01_2021():
+    class TokenIssuanceStartResponse(_abc.IEventResponse):
+        def __init__(self):
+            pass
+                    # super().__init__(kargs)
+                    
+
+    class TokenIssuanceStartData(_abc.IEventData):
+        def __init__(self,
```

Comment:
Can you move these to base class (except for context) and sync them up with the new schema changes.

![image](https://user-images.githubusercontent.com/89389217/156760154-adacceb6-e29e-43df-9783-de11a9fd53d3.png)